x/supplier/keeper/activate_services.go

```diff
@@ -0,0 +1,68 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	proto "github.com/cosmos/gogoproto/proto"
+
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+// BeginBlockerActivateSupplierServices processes suppliers that have pending service activations
+// at the current block height. It returns the number of suppliers whose services were activated.
+func (k Keeper) BeginBlockerActivateSupplierServices(
+	ctx context.Context,
+) (numSuppliersWithServicesActivation uint64, err error) {
+	logger := k.Logger().With("method", "ActivateSupplierServices")
+
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+	currentBlockHeight := sdkCtx.BlockHeight()
+
+	// Only activate supplier services at the start of a session.
+	if !sharedtypes.IsSessionStartHeight(&sharedParams, currentBlockHeight) {
+		return numSuppliersWithServicesActivation, nil
+	}
+
```

Comment:
Can you add a log here: "Starting Session X, about to activate new suppliers..."